### PR TITLE
Addresses #27 adding support for maps the be serialized as json

### PR DESCRIPTION
**Pull request by checketts**

### CODE REVIEW

File: src/main/java/io/github/wimdeblauwe/hsbt/thymeleaf/HtmxAttributeProcessor.java [hunk 1]
```diff
@@ -14,22 +17,25 @@
...
     public static final int ATTR_PRECEDENCE = 1000;
     private final String attrName;
+    private final ObjectMapper mapper;
 
     private static final TemplateMode TEMPLATE_MODE = TemplateMode.HTML;
 
     private AttributeDefinition targetAttributeDefinition;
 
 
     public HtmxAttributeProcessor(final String dialectPrefix,
```
Comment (wimdeblauwe) [inline src/main/java/io/github/wimdeblauwe/hsbt/thymeleaf/HtmxAttributeProcessor.java:34]: Maybe to be consistent, we can put each parameter on a new line. I would also remove the `final` of `dialectPrefix`, I find this is noise for little gain.

File: src/main/java/io/github/wimdeblauwe/hsbt/thymeleaf/HtmxAttributeProcessor.java [hunk 2]
```diff
@@ -52,7 +58,18 @@ protected final void doProcess(
         if (expressionResult == null) {
             structureHandler.removeAttribute(attributeName);
         } else {
-            String newAttributeValue = HtmlEscape.escapeHtml4Xml(expressionResult.toString());
+            String expressionResultString;
+            if (expressionResult instanceof LinkedHashMap) {
+                try {
+                    expressionResultString = this.mapper.writeValueAsString(expressionResult);
+                } catch (JsonProcessingException e) {
+                    throw new TemplateProcessingException("Exception writing map", tag.getTemplateName() ,tag.getLine(), tag.getLine(), e);
```
Comment (wimdeblauwe) [inline src/main/java/io/github/wimdeblauwe/hsbt/thymeleaf/HtmxAttributeProcessor.java:66]:
```suggestion
                    throw new TemplateProcessingException("Exception writing map", tag.getTemplateName(), tag.getLine(), tag.getLine(), e);
```

File: src/test/java/io/github/wimdeblauwe/hsbt/thymeleaf/HtmxDialectTest.java [hunk 3]
```diff
@@ -114,4 +114,14 @@ void testWithThEach() throws Exception {
                 .containsPattern("hx-target-div-with-th-each.*hx-target=\"2\"")
                 .containsPattern("hx-target-div-with-th-each.*hx-target=\"3\"");
     }
+
+    @Test
+    void testWithHxVals() throws Exception {
```
Comment (wimdeblauwe) [inline src/test/java/io/github/wimdeblauwe/hsbt/thymeleaf/HtmxDialectTest.java:119]: Maybe also add a test with multiple values, maybe with a string, a boolean and a number to cover those 3 options?

**Review by wimdeblauwe — CHANGES_REQUESTED** [review]
Looks good in general, a few small remarks. Maybe also add a section to the readme that documents this?

Thanks for this!

**Comment by wimdeblauwe** [comment]
@checketts Are you still working on this? I would like to release a 1.0.0 so I can also then release a 2.0.0 with Spring Boot 3 compatibility next week.

**Review by wimdeblauwe — APPROVED** [review]

**Comment by wimdeblauwe** [comment]
@checketts Could you review the changes I have done to see if they are ok?

**Comment by checketts** [comment]
I'm sorry an email rule was muting some github notifications to a subfolder and I just noticed it. Your changes look great, thanks for adding those test cases!

**Pull request merged**